### PR TITLE
Update applicationType label of all apps when the parent Application Template has an updated applicationType

### PR DESCRIPTION
**Pull request by emmakarova**
**Description**

When application template is updated if it's name has to be changed (in ApplicationTemplateUpdateInput  the name is different from this in the database) all applications that are registered from this app template has to be listed and their applicationType label has to be updated with the new app template name from the input in the mutation.

Changes proposed in this pull request:

**Related issue(s)**

- ...

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [ ] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

**Comment by emmakarova** [comment]
reset-vm

File: components/director/internal/domain/apptemplate/service.go [hunk 1]
```diff
@@ -310,6 +325,27 @@ func (s *service) Update(ctx context.Context, id string, in model.ApplicationTem
...
+	if oldAppTemplate.Name != appTemplate.Name {
+		log.C(ctx).Infof("Listing applications registered from application template with id %s", id)
+		appsByAppTemplate, err := s.appRepo.ListAllByApplicationTemplateID(ctx, id)
+		if err != nil {
+			return errors.Wrapf(err, "while listing applications for app template with id %s", id)
+		}
+
+		for _, app := range appsByAppTemplate {
+			log.C(ctx).Infof("Updating %s label for application with id %s", applicationTypeLabelKey, app.ID)
+			err = s.labelUpsertService.UpsertLabel(ctx, appTenant, &model.LabelInput{
```
[the remark this answers is not in the document]
Comment (emmakarova) [reply components/director/internal/domain/apptemplate/service.go]: Fix: Use upsertLabelGlobal instead upserиtlabel for one tenant

File: components/director/internal/domain/label/service.go [hunk 2]
```diff
@@ -94,6 +98,22 @@ func (s *labelService) UpsertLabel(ctx context.Context, tenant string, labelInpu
...
+	if err := s.validateLabel(ctx, "", labelInput); err != nil {
+		return err
+	}
+
+	label := labelInput.ToLabel(s.uidService.Generate(), "")
+
+	if err := s.labelRepo.UpsertGlobal(ctx, label); err != nil {
+		return errors.Wrapf(err, "while creating Label with id %s for %s with id %s", label.ID, label.ObjectType, label.ObjectID)
+	}
+	log.C(ctx).Debugf("Successfully created Label with id %s for %s with id %s", label.ID, label.ObjectType, label.ObjectID)
```
Comment (kaloyanspiridonov) [inline components/director/internal/domain/label/service.go:112]: created -> upserted ?

File: components/director/internal/domain/label/service.go [hunk 3]
```diff
@@ -94,6 +98,22 @@ func (s *labelService) UpsertLabel(ctx context.Context, tenant string, labelInpu
...
+// UpsertLabelGlobal missing godoc
+func (s *labelService) UpsertLabelGlobal(ctx context.Context, labelInput *model.LabelInput) error {
+	if err := s.validateLabel(ctx, "", labelInput); err != nil {
+		return err
+	}
+
+	label := labelInput.ToLabel(s.uidService.Generate(), "")
+
+	if err := s.labelRepo.UpsertGlobal(ctx, label); err != nil {
+		return errors.Wrapf(err, "while creating Label with id %s for %s with id %s", label.ID, label.ObjectType, label.ObjectID)
```
Comment (kaloyanspiridonov) [inline components/director/internal/domain/label/service.go:110]: creating -> upserting ?

Comment (kaloyanspiridonov) [reply components/director/internal/domain/label/service.go:110]: You can also use %q instead of %s

**Review by kaloyanspiridonov — APPROVED** [review]

**Pull request merged**